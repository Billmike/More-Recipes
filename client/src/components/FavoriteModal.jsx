import React from 'react';
import Modal from 'react-modal';

const FavoriteModal = props => (
  <Modal
    isOpen={!!props.selectedFavorite}
    onRequestClose={props.handleClearFavoriteRecipe}
    closeTimeoutMS={300}
    className="modal-class"
    ariaHideApp={false}
  >
    <h3 className="modal-ptag"> Remove recipe</h3>
    <p className="modal-ptag"> Are you sure want to remove this recipe?</p>
    <div className="group-btn">
      <button
        className="btn btn-danger user-btn modal-btn-red"
        onClick={props.removeFavorite}
      >
        Yes
      </button>
      <button
        className="btn user-btn modal-btn-white"
        onClick={props.handleClearFavoriteRecipe}
      >
        No
      </button>
    </div>
  </Modal>
);

export default FavoriteModal;
