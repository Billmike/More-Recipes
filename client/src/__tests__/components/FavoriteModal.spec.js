import React from 'react';
import { shallow } from 'enzyme';
import FavoriteModal from '../../components/FavoriteModal';

describe('<Modal />', () => {
  it('Should render the Modal correctly', () => {
    const wrapper = shallow(<FavoriteModal />);
    expect(wrapper).toMatchSnapshot();
  });
});
