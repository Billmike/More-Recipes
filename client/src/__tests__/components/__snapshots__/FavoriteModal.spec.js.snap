// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should render the Modal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal-class"
  closeTimeoutMS={300}
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal-ptag"
  >
     Remove recipe
  </h3>
  <p
    className="modal-ptag"
  >
     Are you sure want to remove this recipe?
  </p>
  <div
    className="group-btn"
  >
    <button
      className="btn btn-danger user-btn modal-btn-red"
    >
      Yes
    </button>
    <button
      className="btn user-btn modal-btn-white"
    >
      No
    </button>
  </div>
</Modal>
`;
